analyzers/src/SonarAnalyzer.CSharp/Rules/ParametersCorrectOrder.cs

```diff
@@ -35,6 +35,16 @@ public sealed class ParametersCorrectOrder : ParametersCorrectOrderBase<Argument
     {
         private static readonly DiagnosticDescriptor rule =
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+
+        private static SyntaxToken? GetExpressionSyntaxIdentifier(ExpressionSyntax expressionSyntax) =>
+            expressionSyntax switch
+            {
+                IdentifierNameSyntax identifierNameSyntax => identifierNameSyntax.Identifier,
+                MemberAccessExpressionSyntax memberAccessExpressionSyntax => GetExpressionSyntaxIdentifier(memberAccessExpressionSyntax.Expression),
+                CastExpressionSyntax castExpressionSyntax => GetExpressionSyntaxIdentifier(castExpressionSyntax.Expression),
```

Comment:
Let's add more test cases with some combined nested scenarios. Like
```
object a;
((int?)a).Value
(int)a.Value
```
You might need to use `expression.RemoveParenthesis() switch` and add a test case for that.

Reply:
Have added ParenthesizedExpressionSyntax case